Assets/Scripts/Weapons/Weapon.cs

```diff
@@ -0,0 +1,68 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class Weapon : MonoBehaviour
+{
+    GameObject cameraView;
+
+    Dictionary<string, bool> _firingBehavior = new Dictionary<string, bool>()
+    {
+        { "singleShot", true }, // singleShot == !repeater
+        { "shotgun", false },
+        { "repeater", false }
+    };
+
+    IHitBehaviour _hitBehaviour;
+    float _fireRate; // shots per second
+    int _magazineSize;
+    int _currentMagazine;
+    double lastShot = 0;
+
+    Dictionary<string, bool> _hitTypeModifiers = new Dictionary<string, bool>()
```

Comment:
These modifiers, both hitType and hitStats, should be on the HitBehaviour object.

Reply:
https://github.com/MaXeraph/Adrenarun/pull/4/commits/d65f1bb4fc4d7b6c9c127e9824b438d75c8e0b50#diff-a67bf0084ff9df9f27d4465cf0ef099b6ea3a5b4536f28f160b24bf34eb4f4e4R7-R27